deps/triemap/triemap.c

```diff
@@ -1065,6 +1066,82 @@ int TrieMapIterator_NextContains(TrieMapIterator *it, char **ptr, tm_len_t *len,
   return 0;
 }
 
+int TrieMapIterator_NextWildcard(TrieMapIterator *it, char **ptr, tm_len_t *len, void **value) {
+  if (TimedOut_WithCounter(&it->timeout, &it->timeoutCounter)) {
+    return 0;
+  }
+
+  while (array_len(it->stack) > 0) {
+    __tmi_stackNode *current = __tmi_current(it);
+    TrieMapNode *n = current->n;
+
+    // term string len is equal or longer than fixed query len, trim search branch
+    // children nodes have at least 1 char
+    if (it->mode == TM_WILDCARD_FIXED_LEN_MODE) {
+      int currentNodeLen = current->state == TM_ITERSTATE_SELF ? current->n->len : 1;
+      if (array_len(it->buf) + currentNodeLen > it->prefixLen) {
+        __tmi_Pop(it);
+        goto next;
+      }
+    }
+
+    if (current->state == TM_ITERSTATE_SELF) {
+      // update buffer with current node chars
+      it->buf = array_ensure_append_n(it->buf, current->n->str, current->n->len);
+      current->stringOffset = current->n->len;
+      current->state = TM_ITERSTATE_CHILDREN;
+
+      // check if current buffer is a match
+      int match = current->found ? FULL_MATCH : 
+                  Wildcard_MatchChar(it->prefix, it->prefixLen, it->buf, array_len(it->buf));
+      switch (match) {
+        case NO_MATCH: {
+          __tmi_Pop(it);
+          goto next;
+        }
+        case FULL_MATCH: {
+          // if query string ends with *, all following children are a match
+          if (it->buf[array_len(it->buf) - 1] == '*') {
+            current->found = true;
+          }
+          // current node is a term and should be returned
```

Comment:
```suggestion
          // current node is a terminal and should be returned
```
I guess this is what you meant? Or I got it wrong...